### PR TITLE
LPD-40179 Allow exporting and importing one Page referencing Order Forecasts Chart widget across environments via Headless APIs

### PR DESCRIPTION
**Pull request by michelevigilante**
https://liferay.atlassian.net/browse/LPD-40179

### CODE REVIEW

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:help

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:test:upgrade

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:upgrade

**Comment by michelevigilante** [comment]
ci:report:84351081

**Comment by michelevigilante** [comment]
ci:report:84387055

**Comment by michelevigilante** [comment]
ci:report:84387091

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:report:84573279

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java [hunk 1]
```diff
@@ -39,11 +46,54 @@ public CommerceDashboardForecastDisplayContext(
 			ConfigurationProviderUtil.getPortletInstanceConfiguration(
 				CommerceDashboardForecastPortletInstanceConfiguration.class,
 				_commerceDashboardForecastRequestHelper.getThemeDisplay());
+
+		_themeDisplay =
+			_commerceDashboardForecastRequestHelper.getThemeDisplay();
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/display/context/CommerceDashboardForecastDisplayContext.java:51]: I think you don't need it. You can get the same info from `_commerceDashboardForecastRequestHelper`

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java [hunk 2]
```diff
@@ -52,7 +55,7 @@ public void include(
 				commerceDashboardForecastDisplayContext =
 					new CommerceDashboardForecastDisplayContext(
 						_accountEntryModelResourcePermission,
-						httpServletRequest);
+						httpServletRequest, _assetCategoryLocalService);
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java:58]: Sort

File: modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/build.gradle [hunk 3]
```diff
@@ -14,6 +14,7 @@ dependencies {
 	compileOnly project(":apps:commerce:commerce-product-api")
 	compileOnly project(":apps:dispatch:dispatch-api")
 	compileOnly project(":apps:portal:portal-aop-api")
+	compileOnly project(":apps:static:portal:portal-upgrade-api")
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/build.gradle:17]: Move upgrade process into `commerce-dashboard-web`

File: modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/src/main/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/registry/CommerceMachineLearningForecastServiceUpgradeStepRegistrator.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/**
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/src/main/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/registry/CommerceMachineLearningForecastServiceUpgradeStepRegistrator.java:1]: Move upgrade process into `commerce-dashboard-web`

File: modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/bnd.bnd [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Commerce Machine Learning Forecast Alert Service
 Bundle-SymbolicName: com.liferay.commerce.machine.learning.forecast.alert.service
 Bundle-Version: 5.0.39
-Liferay-Require-SchemaVersion: 1.0.0
+Liferay-Require-SchemaVersion: 1.0.1
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/bnd.bnd:4]: Rollback

File: modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/src/main/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/CommerceMachineLearningForecastPortletPreferencesUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,102 @@
+/**
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-forecast-alert-service/src/main/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/CommerceMachineLearningForecastPortletPreferencesUpgradeProcess.java:1]: Move upgrade process into `commerce-dashboard-web`

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardForecastsChartPortlet.java [hunk 7]
```diff
@@ -7,8 +7,8 @@
 
 import com.liferay.account.model.AccountEntry;
 import com.liferay.asset.kernel.service.AssetCategoryLocalService;
-import com.liferay.commerce.machine.learning.internal.constants.CommerceDashboardPortletKeys;
 import com.liferay.commerce.dashboard.web.internal.display.context.CommerceDashboardForecastDisplayContext;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardForecastsChartPortlet.java:11]: Rollback

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardForecastsChartPortletProvider.java [hunk 8]
```diff
@@ -5,7 +5,7 @@
 
 package com.liferay.commerce.dashboard.web.internal.portlet;
 
-import com.liferay.commerce.dashboard.web.internal.constants.CommerceDashboardPortletKeys;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardForecastsChartPortletProvider.java:8]: Rollback

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardHistoryChartPortlet.java [hunk 9]
```diff
@@ -5,7 +5,7 @@
 
 package com.liferay.commerce.dashboard.web.internal.portlet;
 
-import com.liferay.commerce.dashboard.web.internal.constants.CommerceDashboardPortletKeys;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardHistoryChartPortlet.java:8]: Rollback

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardOverviewChartPortlet.java [hunk 10]
```diff
@@ -5,7 +5,7 @@
 
 package com.liferay.commerce.dashboard.web.internal.portlet;
 
-import com.liferay.commerce.dashboard.web.internal.constants.CommerceDashboardPortletKeys;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardOverviewChartPortlet.java:8]: Rollback

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardStatusChartPortlet.java [hunk 11]
```diff
@@ -5,7 +5,7 @@
 
 package com.liferay.commerce.dashboard.web.internal.portlet;
 
-import com.liferay.commerce.dashboard.web.internal.constants.CommerceDashboardPortletKeys;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/CommerceDashboardStatusChartPortlet.java:8]: Rollback

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java [hunk 12]
```diff
@@ -9,7 +9,7 @@
 import com.liferay.asset.kernel.model.AssetCategory;
 import com.liferay.asset.kernel.service.AssetCategoryLocalService;
 import com.liferay.commerce.dashboard.web.internal.display.context.CommerceDashboardForecastDisplayContext;
-import com.liferay.commerce.machine.learning.internal.constants.CommerceDashboardPortletKeys;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/portlet/action/CommerceDashboardForecastConfigurationAction.java:12]: Rollback

File: modules/dxp/apps/commerce/commerce-machine-learning-api/bnd.bnd [hunk 13]
```diff
@@ -2,5 +2,6 @@ Bundle-Name: Liferay Commerce Machine Learning API
 Bundle-SymbolicName: com.liferay.commerce.machine.learning.api
 Bundle-Version: 7.2.6
 Export-Package:\
+	com.liferay.commerce.machine.learning.constants,\
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-api/bnd.bnd:5]: Rollback

File: modules/dxp/apps/commerce/commerce-machine-learning-api/build.gradle [hunk 14]
```diff
@@ -1,4 +1,5 @@
 dependencies {
 	compileOnly group: "biz.aQute.bnd", name: "biz.aQute.bndlib", version: "3.1.0"
 	compileOnly group: "com.liferay.portal", name: "com.liferay.portal.kernel", version: "default"
+	compileOnly project(":apps:commerce:commerce-api")
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-api/build.gradle:4]: Rollback

File: modules/dxp/apps/commerce/commerce-machine-learning-api/src/main/java/com/liferay/commerce/machine/learning/constants/CommerceDashboardPortletKeys.java [hunk 15]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-package com.liferay.commerce.dashboard.web.internal.constants;
+package com.liferay.commerce.machine.learning.constants;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-api/src/main/java/com/liferay/commerce/machine/learning/constants/CommerceDashboardPortletKeys.java:6]: Rollback

File: modules/dxp/apps/commerce/commerce-machine-learning-api/src/main/resources/com/liferay/commerce/machine/learning/constants/packageinfo [hunk 16]
```diff
@@ -0,0 +1 @@
+version 1.0.0
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-api/src/main/resources/com/liferay/commerce/machine/learning/constants/packageinfo:1]: Drop commit instead

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/exportimport/portlet/preferences/processor/CommerceDashboardForecastExportImportPortletPreferencesProcessor.java [hunk 17]
```diff
@@ -0,0 +1,66 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.commerce.dashboard.web.internal.exportimport.portlet.preferences.processor;
+
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/exportimport/portlet/preferences/processor/CommerceDashboardForecastExportImportPortletPreferencesProcessor.java:8]: Remember to change portlet key

File: modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/exportimport/portlet/preferences/processor/CommerceDashboardForecastExportImportPortletPreferencesProcessor.java [hunk 18]
```diff
@@ -0,0 +1,66 @@
...
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Michele Vigilante
+ */
+@Component(
+	property = "javax.portlet.name=" + CommerceDashboardPortletKeys.COMMERCE_DASHBOARD_FORECASTS_CHART,
+	service = ExportImportPortletPreferencesProcessor.class
+)
+public class CommerceDashboardForecastExportImportPortletPreferencesProcessor
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-dashboard-web/src/main/java/com/liferay/commerce/dashboard/web/internal/exportimport/portlet/preferences/processor/CommerceDashboardForecastExportImportPortletPreferencesProcessor.java:29]: `CommerceDashboardForecast` -> `CommerceDashboardForecastsChart`

File: modules/dxp/apps/commerce/commerce-machine-learning-test/build.gradle [hunk 19]
```diff
@@ -2,15 +2,19 @@ dependencies {
...
 	testIntegrationImplementation project(":apps:commerce:commerce-api")
 	testIntegrationImplementation project(":apps:commerce:commerce-currency-api")
 	testIntegrationImplementation project(":apps:commerce:commerce-currency-test-util")
 	testIntegrationImplementation project(":apps:commerce:commerce-product-api")
 	testIntegrationImplementation project(":apps:commerce:commerce-product-test-util")
 	testIntegrationImplementation project(":apps:commerce:commerce-test-util")
+	testIntegrationImplementation project(":apps:export-import:export-import-test-util")
 	testIntegrationImplementation project(":apps:info:info-api")
 	testIntegrationImplementation project(":apps:portal-search:portal-search-test-util")
+	testIntegrationImplementation project(":dxp:apps:commerce:commerce-dashboard-web")
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/build.gradle:20]: commerce-dashboard-web doesn't export packages. This dependency is useless.

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+ */
+
+package com.liferay.commerce.machine.learning.forecast.exportimport.portlet.preferences.processor.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.asset.kernel.model.AssetCategory;
+import com.liferay.asset.kernel.model.AssetVocabulary;
+import com.liferay.asset.kernel.service.AssetCategoryLocalServiceUtil;
+import com.liferay.asset.test.util.AssetTestUtil;
+import com.liferay.commerce.machine.learning.constants.CommerceDashboardPortletKeys;
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java:13]: Remove import

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java [hunk 21]
```diff
@@ -0,0 +1,139 @@
...
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Override
+	public String getPortletId() throws Exception {
+		return PortletIdCodec.encode(
+			CommerceDashboardPortletKeys.COMMERCE_DASHBOARD_FORECASTS_CHART,
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java:49]: Use the string directly

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java [hunk 22]
```diff
@@ -0,0 +1,139 @@
...
+			group.getGroupId(), assetVocabulary.getVocabularyId());
+
+		PortletPreferences portletPreferences = getImportedPortletPreferences(
+			HashMapBuilder.put(
+				"assetCategoryExternalReferenceCodes",
+				() -> new String[] {assetCategory.getExternalReferenceCode()}
+			).build());
+
+		AssetCategory importedAssetCategory =
+			AssetCategoryLocalServiceUtil.
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java:83]: I don't want to see Utils anymore, please Inject the service.

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+			group2.getGroupId(), assetVocabulary.getVocabularyId());
+
+		PortletPreferences portletPreferences = getImportedPortletPreferences(
+			HashMapBuilder.put(
+				"assetCategoryExternalReferenceCodes",
+				() -> new String[] {assetCategory.getExternalReferenceCode()}
+			).build());
+
+		AssetCategory importedAssetCategory =
+			AssetCategoryLocalServiceUtil.
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/exportimport/portlet/preferences/processor/test/CommerceDashboardForecastsChartExportImportTest.java:123]: Same here

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Michele Vigilante
+ */
+@RunWith(Arquillian.class)
+public class
+	CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest {
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java:45]: `CommerceMachineLearningForecastPortlet` -> `CommerceDashboardForecastsChartPortlet`

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java [hunk 25]
```diff
@@ -0,0 +1,136 @@
...
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpdateCommerceMachineLearningForecastPortletPreferencesExternalReferenceCode()
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java:53]: Change test names as well

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java [hunk 26]
```diff
@@ -0,0 +1,136 @@
...
+
+	@Inject
+	private static PortletLocalService _portletLocalService;
+
+	@Inject
+	private static PortletPreferencesLocalService
+		_portletPreferencesLocalService;
+
+	@Inject(
+		filter = "(&(component.name=com.liferay.commerce.machine.learning.forecast.alert.internal.upgrade.registry.CommerceMachineLearningForecastServiceUpgradeStepRegistrator))"
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java:132]: `CommerceMachineLearningForecastService` -> `CommerceDashboardWeb`

Pay attention to the package as well when you move it.

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java [hunk 27]
```diff
@@ -0,0 +1,136 @@
...
+		String defaultPreferences = StringBundler.concat(
+			"<portlet-preferences><preference><name>assetCategoryIds",
+			"</name><value>", assetCategory.getCategoryId(),
+			"</value></preference></portlet-preferences>");
+
+		PortletPreferences portletPreferences =
+			_portletPreferencesLocalService.addPortletPreferences(
+				group.getCompanyId(), PortletKeys.PREFS_OWNER_ID_DEFAULT,
+				PortletKeys.PREFS_OWNER_TYPE_LAYOUT, layout.getPlid(),
+				CommerceDashboardPortletKeys.COMMERCE_DASHBOARD_FORECASTS_CHART,
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java:82]: Use the string directly

File: modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java [hunk 28]
```diff
@@ -0,0 +1,136 @@
...
+				_portletLocalService.fetchPortletById(
+					group.getCompanyId(),
+					CommerceDashboardPortletKeys.
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest.java:85]: Use the string directly

File: modules/dxp/apps/commerce/source-formatter-suppressions.xml [hunk 29]
```diff
@@ -2,6 +2,7 @@
 
 <suppressions>
 	<source-check>
+		<suppress checks="JavaLongLinesCheck" files="commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/forecast/alert/internal/upgrade/v1_0_1/test/CommerceMachineLearningForecastPortletPreferencesUpgradeProcessTest\.java" />
```
Comment (alessiorendina) [inline modules/dxp/apps/commerce/source-formatter-suppressions.xml:5]: Check if the new name solves this

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:upgrade

**Comment by michelevigilante** [comment]
ci:test:commerce-functional

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:stop

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by michelevigilante** [comment]
ci:report:87562390

**Comment by michelevigilante** [comment]
ci:test:upgrade

**Comment by michelevigilante** [comment]
ci:test:sf

**Comment by michelevigilante** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:forward

**Pull request closed without merging**